File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& A) {
        int count = 1;
        int majority_element = A[0];
        for (int i = 1 ; i < A.size(); i++){
            if (count == 0){
                majority_element = A[i];
                count++;
            } else if (A[i] == majority_element){
                count++;
            } else {
                count--;
            }
        }
        return majority_element;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        cnt = 0
        el = -1
        for n in nums:
            if not cnt:
                el = n
                cnt += 1
            elif n == el:
                cnt += 1
            else:
                cnt -= 1
                
        return el
        